### PR TITLE
Delete hash if destination does not exist

### PR DESCRIPTION
**Pull request by keatonwalker**
Removed a few problems and added a test.

### CODE REVIEW

File: src/forklift/core.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def remove_temp_table(table):
         if not arcpy.Exists(crate.destination):
             log.debug('%s does not exist. creating', crate.destination)
             _create_destination_data(crate)
+            if arcpy.Exists(path.join(hash_gdb_path, crate.name)):
```
Comment (steveoh) [inline src/forklift/core.py]: in what scenario does the crate destination not exist but the table would already exist in the hash table?

File: src/forklift/core.py [hunk 2]
```diff
@@ -94,7 +96,8 @@ def remove_temp_table(table):
         changes = _hash(crate, hash_gdb_path)
 
         #: delete unaccessed hashes
-        if changes.has_deletes:
+        if changes.has_deletes():
```
Comment (steveoh) [inline src/forklift/core.py]: good catch and i enjoy logging 👍

File: tests/test_core.py [hunk 3]
```diff
@@ -62,6 +62,14 @@ def test_update_no_existing_destination(self):
         self.assertEqual(core.update(crate, lambda x: True)[0], Crate.CREATED)
         self.assertEqual(arcpy.Exists(crate.destination), True)
 
+    def test_update_no_deleted_destination(self):
```
Comment (steveoh) [inline tests/test_core.py]: i should be able to understand what this test should do by the name. Can you be more verbose with what this test is testing so I don't have to read the code?

File: src/forklift/core.py (hunk 1 above)
Comment (keatonwalker) [inline src/forklift/core.py]: If the update has happened before, then the destination data gets deleted, the hash will still exist and the destination to hash primary key will be invalid.

This probably shouldn't ever happen. Although, I did it to myself when I was playing with forklift.

File: tests/test_core.py (hunk 3 above)
Comment (keatonwalker) [inline tests/test_core.py]: No problem. I will try for a better name.

**Pull request merged**